core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -216,7 +261,16 @@ public final void accumulate(FieldStats other) {
         isAggregatable |= other.isAggregatable;
 
         assert type == other.getType();
-        updateMinMax((T) other.minValue, (T) other.maxValue);
+        if (other.hasMinMax == false) {
+            return;
+        }
+        if (hasMinMax == false) {
+            hasMinMax = true;
+            minValue = (T) other.minValue;
+            maxValue = (T) other.maxValue;
```

Comment:
it feels wrong to say that we know the min/max unless the information is available for all indices?